dashboard/app/models/course.rb

```diff
@@ -254,17 +245,29 @@ def self.has_any_course_experiments?(user)
 
   # Get the set of valid courses for the dropdown in our sections table, using
   # any alternate scripts based on any experiments the user belongs to.
-  def self.courses_for_user_with_experiments(user)
-    Course.
-      all.
-      select {|course| ScriptConstants.script_in_category?(:full_course, course[:name])}.
+  def self.valid_courses_without_cache(user: nil, include_unstable: false)
+    course_infos = Course.
+      where(name: ScriptConstants::CATEGORIES[:full_course]).
       map {|course| course.assignable_info(user)}
+
+    # For now, infer whether the course is stable from its version year.
+    # * Currently, only 2017 versions are stable.
+    # * With the 2018-teacher-experience experiment, all course versions are stable.
```

Comment:
Here you mention a particular behavior for the `2018-teacher-experience` experiment but I don't see that behavior captured in the implementation of this method.  If this is a hint about a possible calling case for this method, maybe be more explicit about what parameters you'd expect in this case? (e.g. With the experiment enabled, since all versions are stable, do we expect `include_unstable` to be false?)

Reply:
Great point Brad, this comment is out of place. I was mentioning the 2018-teacher-experience so that I'd remember to update this block of code when we launch the experiment, but now I realize that will be done in some caller of this method. I've removed this line of comment and expanded the comment in teacherSectionRedux, which will ultimately control whether `include_unstable` is set here.